lib/ServiceClient.js

```diff
@@ -160,7 +158,32 @@ class ServiceClient extends EventEmitter {
     }
   }
 
+  _initiateServiceConnection(call) {
+    // if we haven't connected to the service yet, create the connection
+    // this will always be the case unless this is persistent service client
+    // calling for a second time.
+    if (!this.getPersist() || this._serviceClient === null) {
+      return this._nodeHandle.lookupService(this.getService())
+      .then((resp) => {
+        const serviceUri = resp[2];
```

Comment:
In most [ROS XMLRPC Api](http://wiki.ros.org/ROS/Master_Slave_APIs) calls the useful information you want is stored in index 2 of the response (index 0 is status code and index 1 is status message). Filtering these responses nicely has been on the backburner for a while...